shell/src/main/java/alluxio/cli/ValidateEnv.java

```diff
@@ -0,0 +1,392 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.master.AlluxioMaster;
+import alluxio.util.network.NetworkAddressUtils;
+import alluxio.util.network.NetworkAddressUtils.ServiceType;
+import alluxio.worker.AlluxioWorker;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.TreeMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utility for checking Alluxio environment.
+ */
+public final class ValidateEnv {
+  private static final String USAGE = "USAGE: ValidateEnv TARGET [NAME]\n\n"
+      + "Validate environment for Alluxio.\n\n"
+      + "TARGET can be one of the following values:\n"
+      + "local: \trun all validation tasks on local\n"
+      + "master: \trun master validation tasks on local\n"
+      + "worker: \trun worker validation tasks on local\n"
+      + "all: \trun corresponding validation tasks on all master nodes and worker nodes\n"
+      + "masters: \trun master validation tasks on all master nodes\n"
+      + "workers: \trun worker validation tasks on all worker nodes\n\n"
+      + "NAME can be any task full name or prefix.\n"
+      + "When NAME is given, only tasks with name starts with the prefix will run.\n";
+
+  private static final Options OPTIONS = new Options();
+
+  private static final Map<ValidationTask, String> TASK_NAMES = new HashMap<>();
+
+  // port availability validations
+  private static final ValidationTask MASTER_RPC_VALIDATION_TASK = create(
+      "master.rpc.port.available",
+      new PortAvailabilityValidationTask(ServiceType.MASTER_RPC, AlluxioMaster.class));
```

Comment:
as per my comment above, I suggest you replace `AlluxioMaster.class` with `"Alluxio Master"` ... and similarly for `AlluxioWorker.class`, replacing it with `"Alluxio Worker"`